src/main/c/Jep/jep_numpy.c

```diff
@@ -589,6 +588,9 @@ PyObject* convert_jdndarray_pyndarray(JNIEnv *env, PyObject* pyobj)
     // get the primitive array and convert it
     data = (*env)->CallObjectMethod(env, obj, dndarrayGetData);
     if (process_java_exception(env) || !data) {
+        free(dims);
+        //Is this necessary?
+        (*env)->DeleteLocalRef(env, data);
```

Comment:
Good catch on free(dims).  I'm not sure what happens if you `(*env)->DeleteLocalRef(env, NULL);`.  It could be bad?  If we don't know we shouldn't add that line (and the comment).

Reply:
I tested it, if I pass NULL to DeleteLocalRef, nothing happens (at least on Java 14). I will remove this unnecessary call.